sources/api/migration/migration-helpers/src/common_migrations.rs

```diff
@@ -152,6 +153,206 @@ mod test_add_prefixes_migration {
     }
 }
 
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+pub struct PrefixSuffix {
+    pub prefix: &'static str,
+    pub suffix: &'static str,
+}
+pub struct AddPrefixSuffixMigration(pub Vec<PrefixSuffix>);
+
+impl Migration for AddPrefixSuffixMigration {
+    /// New versions must either have a default for the settings or generate them; we don't need to
+    /// do anything.
+    fn forward(&mut self, input: MigrationData) -> Result<MigrationData> {
+        println!(
+            "AddPrefixSuffixMigration({:?}) has no work to do on upgrade.",
+            self.0
+                .iter()
+                .map(|ps| format!("{}*{}", ps.prefix, ps.suffix))
+                .collect::<Vec<_>>()
+        );
+        Ok(input)
+    }
+
+    /// Older versions don't know about the settings; we remove them so that old versions don't see
+    /// them and fail deserialization.
+    fn backward(&mut self, mut input: MigrationData) -> Result<MigrationData> {
+        let mut compiled_patterns = Vec::new();
+        for pattern in &self.0 {
+            let regex_pattern = format!(
+                r"^{}\.(.+)\.{}$",
+                regex::escape(pattern.prefix),
+                regex::escape(pattern.suffix)
+            );
+            let regex =
+                Regex::new(&regex_pattern).context(error::InvalidPrefixSuffixPatternSnafu {
+                    prefix: pattern.prefix,
+                    suffix: pattern.suffix,
+                })?;
+            compiled_patterns.push(regex);
+        }
```

Comment:
We have to compile the regex patterns at runtime because it depends on the prefix and suffix. But this way we only still compile it once for the backward() call.